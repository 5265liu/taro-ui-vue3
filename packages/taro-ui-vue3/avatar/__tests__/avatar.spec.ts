import { mount } from '@vue/test-utils'
import AtAvatar from '../src/index.vue'

const factory = (values = {}, slots = { default: [] }) => {
  return mount(AtAvatar, {
    slots,
    props: { ...values },
  })
}

describe('Avatar Snap', () => {
  it('render Avatar -- props size(large) ', () => {
    const wrapper = factory({ size: 'large' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Avatar -- props size(normal) ', () => {
    const wrapper = factory({ size: 'normal' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Avatar -- props size(small) ', () => {
    const wrapper = factory({ size: 'small' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Avatar -- props circle', () => {
    const wrapper = factory({ circle: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Avatar -- props image', () => {
    const wrapper = factory({ image: 'https://jdc.jd.com/img/100' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Avatar -- props text', () => {
    const wrapper = factory({ text: '凹凸实验室' })
    expect(wrapper.element).toMatchSnapshot()
  })
})
