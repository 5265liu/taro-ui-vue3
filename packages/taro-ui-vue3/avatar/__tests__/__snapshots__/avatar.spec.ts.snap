// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Snap render Avatar -- props circle 1`] = `
<view
  class="at-avatar--normal at-avatar--circle at-avatar"
>
  <text
    class="at-avatar__text"
  />
</view>
`;

exports[`Avatar Snap render Avatar -- props image 1`] = `
<view
  class="at-avatar--normal at-avatar"
>
  <image
    class="at-avatar__img"
    src="https://jdc.jd.com/img/100"
  />
</view>
`;

exports[`Avatar Snap render Avatar -- props size(large)  1`] = `
<view
  class="at-avatar--large at-avatar"
>
  <text
    class="at-avatar__text"
  />
</view>
`;

exports[`Avatar Snap render Avatar -- props size(normal)  1`] = `
<view
  class="at-avatar--normal at-avatar"
>
  <text
    class="at-avatar__text"
  />
</view>
`;

exports[`Avatar Snap render Avatar -- props size(small)  1`] = `
<view
  class="at-avatar--small at-avatar"
>
  <text
    class="at-avatar__text"
  />
</view>
`;

exports[`Avatar Snap render Avatar -- props text 1`] = `
<view
  class="at-avatar--normal at-avatar"
>
  <text
    class="at-avatar__text"
  >
    å‡¹
  </text>
</view>
`;
